core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpMethodRepoDAO.scala

```diff
@@ -59,18 +60,25 @@ class HttpMethodRepoDAO(baseMethodRepoServiceURL: String, apiPath: String = "",
   }
 
   override def getMethodConfig( namespace: String, name: String, version: Int, userInfo: UserInfo ): Future[Option[AgoraEntity]] = {
-    getAgoraEntity(s"${methodRepoServiceURL}/configurations/${namespace}/${name}/${version}",userInfo)
+    getAgoraEntity(s"$agoraServiceURL/configurations/$namespace/$name/$version",userInfo)
   }
 
   override def getMethod( method: MethodRepoMethod, userInfo: UserInfo ): Future[Option[AgoraEntity]] = {
     method match {
       case agoraMethod: AgoraMethod =>
-        getAgoraEntity(s"${methodRepoServiceURL}/methods/${agoraMethod.methodNamespace}/${agoraMethod.methodName}/${agoraMethod.methodVersion}", userInfo)
-      case _ =>
-        throw new RawlsException(s"Method repo '${method.repo.scheme}' not yet supported")
+        getAgoraEntity(s"$agoraServiceURL/methods/${agoraMethod.methodNamespace}/${agoraMethod.methodName}/${agoraMethod.methodVersion}", userInfo)
+      case dockstoreMethod: DockstoreMethod =>
+        getDockstoreEntity(dockstoreMethod) flatMap { response: GA4GHToolDescriptor =>
+          // TODO: re-using AgoraEntity feels sketchy to me. It seems to work without any changes, but should we create a DockstoreEntity?
+          Future(Some(AgoraEntity(payload = Some(response.descriptor), entityType = Some(AgoraEntityType.Workflow))))
+        }
     }
   }
 
+  private def getDockstoreEntity(method: DockstoreMethod): Future[GA4GHToolDescriptor] = {
+    httpClientUtils.executeRequestUnmarshalResponse[GA4GHToolDescriptor](http, Get(method.ga4ghDescriptorUrl(dockstoreServiceURL)))
+  }
```

Comment:
`getAgoraEntity` is called what it is because the URL may specify either `/methods/` or `/configurations/`.

In light of the above observation, I renamed `getDockstoreEntity` to `getDockstoreMethod` for clarity.

Because we are calling public Dockstore APIs, I created an unauthenticated version of `pipeline` so that we can have `getDockstoreMethod` and ` getAgoraEntity` make their HTTP calls at the same level of abstraction.